### PR TITLE
Initial support for expension of CSS font shorthand property

### PR DESCRIPTION
**Pull request by mkiael**
This PR adds support for parsing font size and family.

### CODE REVIEW

File: css/parser_test.cpp [hunk 1]
```diff
@@ -217,5 +217,65 @@ int main() {
...
+        expect(body.declarations.at("font-size"s) == "10%"s);
+        expect(body.declarations.at("font-stretch"s) == "normal"s);
+        expect(body.declarations.at("font-style"s) == "normal"s);
+        expect(body.declarations.at("font-variant"s) == "normal"s);
+        expect(body.declarations.at("font-weight"s) == "normal"s);
+        expect(body.declarations.at("line-height"s) == "2.5"s);
+    });
+
+    etest::test("parser: shorthand font with absolute size, line height, and font family", [] {
+        auto rules = css::parse("p { font: x-large/110% \"New Century Schoolbook\", serif; }"sv);
```
Comment (robinlinden) [inline css/parser_test.cpp:266]: Could be done with raw string literals to avoid the escaping.

File: css/parser.h [hunk 2]
```diff
@@ -9,7 +9,9 @@
 
 #include "util/base_parser.h"
 
+#include <array>
 #include <cstring>
+#include <optional>
```
Comment (robinlinden) [inline css/parser.h:14]: This belongs to the previous commit?

**Review by robinlinden — APPROVED** [review]
Looks great! Up to you if you want to fix the minor things I commented on or merge (ff-only!). :P Some of them are more related to commit hygiene than the changes anyway.

I also think you should add your name to the copyright headers in `css/parser.h` and `css/parser_test.cpp`!

Just a something like `SPDX-FileCopyrightText: 2021 mkiael <mkiael@realema.il>` under the existing copyright line. :P

**Comment by mkiael** [comment]
Thanks for the review! I'll fix your comments before merging this.

**Pull request merged**